rq/worker.py

```diff
@@ -550,7 +557,9 @@ def heartbeat(self, timeout=None, pipeline=None):
         timeout = timeout or self.default_worker_ttl
         connection = pipeline if pipeline is not None else self.connection
         connection.expire(self.key, timeout)
-        connection.hset(self.key, 'last_heartbeat', utcformat(utcnow()))
+        result = connection.hset(self.key, 'last_heartbeat', utcformat(utcnow()))
```

Comment:
This is most probably sent within a Redis pipeline so you'll need to check the result after `pipeline.execute()` is called.